demo_nodes_cpp/CMakeLists.txt

```diff
@@ -159,7 +159,7 @@ if(BUILD_TESTING)
   # Add each test case.  Multi-executable tests can be specified in
   # semicolon-separated strings, like  exe1;exe2.
   set(tutorial_tests
-    "content_filtering_publisher:content_filtering_subscriber"
+    "content_filtering_publisher@publish_ms=100:content_filtering_subscriber"
```

Comment:
Yeah, I know exactly what you mean.  I was actually hoping to do this parsing strictly in Python, but I couldn't do that because I need to know the executable name in CMakeLists.txt below.

One other idea on how to do this would be to have a separate variable which is just the parameters that correspond to a particular executable.  That is, if the executable is `content_filtering_publisher`, then we'll look for parameters in the `${content_filtering_publisher_PARAMETERS}` variable if and only if it exists.  It is more magical in that you just have to know we'll look for that, but it should be less parsing code.  Do you think that is better?

Reply:
I think I prefer what you have currently over specific variable names.  That seems to be even trickier and have the potential to have hidden side effects.